### PR TITLE
Legacy controller execution flow : Changed a png diagram to a mermaid one to improve searchability and maintainability

### PR DESCRIPTION
**Pull request by thomasnares**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       |8.x
| Description?  | Changed a png diagram to a mermaid one to improve searchability and maintainability

### CODE REVIEW

**Review by 0x346e3730 — APPROVED** [review]

File: development/architecture/legacy/legacy-controllers.md [hunk 1]
```diff
@@ -10,4 +11,40 @@ Legacy controllers are based on PrestaShop's custom framework and go a long way
 
 Legacy controllers work best when a Controller performs a single action, for example, render a page. The process has been divided in several methods, which simplifies customization via method override.
 
-{{< figure src="../img/legacy-controller-flow.png" title="Execution flow of legacy controllers" >}}
+<div class='mermaid'>
+%%{ init: { 'flowchart': { 'curve': 'stepAfter' } } }%%
+graph TB
+    A(("Controller::run()")) --> B("init()<br><i><small>Initializes the controller</small></i>")    
+    B:::importantStep --> C{"checkAccess()<br><i><small>Check if the controller<br>is available for the<br> current user/viistor</small></i>"}
```
Comment (kpodemski) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    B:::importantStep --> C{"checkAccess()<br><i><small>Check if the controller<br>is available for the<br> current user/vistor</small></i>"}
```

File: development/architecture/legacy/legacy-controllers.md [hunk 2]
```diff
@@ -10,4 +11,40 @@ Legacy controllers are based on PrestaShop's custom framework and go a long way
...
+    C:::decision -- false --> D("initCursedPage()<br><i><small>Assigns Smarty variables when access is forbidden</small></i>")
+    D --> E("smartyOutputContent()<br><i><small>Display page content</small></i>")
```
Comment (kpodemski) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    D --> E("smartyOutputContent()<br><i><small>Displays the page content</small></i>")
```

File: development/architecture/legacy/legacy-controllers.md [hunk 3]
```diff
@@ -10,4 +11,40 @@ Legacy controllers are based on PrestaShop's custom framework and go a long way
...
+    E:::importantStep
+    C -- true --> G("setMedia()<br><i><small>Sets controller CSS and JS files</small></i>")
```
Comment (kpodemski) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    C -- true --> G("setMedia()<br><i><small>Sets controller's CSS and JS files</small></i>")
```

File: development/architecture/legacy/legacy-controllers.md [hunk 4]
```diff
@@ -10,4 +11,40 @@ Legacy controllers are based on PrestaShop's custom framework and go a long way
...
+    G:::notAlwaysRun --> H("postProcess()<br><i><small>Used to process user input</small></i>")
+    H:::importantStep --> I{"has redirect_after"}
+    I:::decision -- true --> J("redirect()<br><i><small>Redirects to redirect_after after<br>the process if there is no error</small></i>")
```
Comment (kpodemski) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    I:::decision -- true --> J("redirect()<br><i><small>If there is no error, redirects after the process to the "redirect_after"</small></i>")
```

File: development/architecture/legacy/legacy-controllers.md [hunk 5]
```diff
@@ -10,4 +11,40 @@ Legacy controllers are based on PrestaShop's custom framework and go a long way
...
+    I -- false --> K("initHeader()<br><i><small>Assigns Smarty variables<br>for the page header</small></i>")
+    J --> K
+    K:::notAlwaysRun --> L{"viewAccess()<br><i><small>Check if the current user/visiter<br>has valid view permissions</small></i>"}
```
Comment (kpodemski) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    K:::notAlwaysRun --> L{"viewAccess()<br><i><small>Checks if the current user/visitor<br>has valid view permissions</small></i>"}
```

File: development/architecture/legacy/legacy-controllers.md [hunk 6]
```diff
@@ -10,4 +11,40 @@ Legacy controllers are based on PrestaShop's custom framework and go a long way
...
+    L:::decision -- false --> M("Add access denied error message<br><i><small>Added to errors property</small></i>")
+    L -- true --> N("initContent()<br><i><small>Assigns Smarty variables<br>for the page main content</small></i>")
```
Comment (kpodemski) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    L -- true --> N("initContent()<br><i><small>Assigns Smarty variables<br>for the main content of the page</small></i>")
```

**Review by kpodemski — COMMENTED** [review]
A few comments :)

File: development/architecture/legacy/legacy-controllers.md (hunk 1 above)
Comment (thomasnares) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    B:::importantStep --> C{"checkAccess()<br><i><small>Check if the controller<br>is available for the<br> current user/visitor</small></i>"}
```

Comment (thomasnares) [reply development/architecture/legacy/legacy-controllers.md]: better like this :D

File: development/architecture/legacy/legacy-controllers.md (hunk 4 above)
Comment (thomasnares) [reply development/architecture/legacy/legacy-controllers.md]: i need to escape the "

Comment (thomasnares) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    I:::decision -- true --> J("redirect()<br><i><small>If there is no error, redirects after the process to the &quot;redirect_after&quot;</small></i>")
```

File: development/architecture/legacy/legacy-controllers.md [hunk 7]
```diff
@@ -10,4 +11,40 @@ Legacy controllers are based on PrestaShop's custom framework and go a long way
...
+graph TB
+    A(("Controller::run()")) --> B("init()<br><i><small>Initializes the controller</small></i>")    
+    B:::importantStep --> C{"checkAccess()<br><i><small>Check if the controller<br>is available for the<br> current user/visitor</small></i>"}
+    C:::decision -- false --> D("initCursedPage()<br><i><small>Assigns Smarty variables when access is forbidden</small></i>")
+    D --> E("smartyOutputContent()<br><i><small>Displays the page content</small></i>")
+    E:::importantStep
+    C -- true --> G("setMedia()<br><i><small>Sets controller's CSS and JS files</small></i>")
+    G:::notAlwaysRun --> H("postProcess()<br><i><small>Used to process user input</small></i>")
+    H:::importantStep --> I{"has redirect_after"}
+    I:::decision -- true --> J("redirect()<br><i><small>If there is no error, redirects after the process to the &quot;redirect_after&quot;</small></i>")
```
Comment (thomasnares) [inline development/architecture/legacy/legacy-controllers.md]:
```suggestion
    I:::decision -- true --> J("redirect()<br><i><small>If there is no error, redirects after the process to the #quot;redirect_after#quot;</small></i>")
```

**Review by kpodemski — APPROVED** [review]

**Pull request merged**